src/main/java/seam/SeamCalculator.java

```diff
@@ -31,116 +31,68 @@ public Pixel[][] calculateAndRemoveSeams()
         for (int verticalSeam = 0; verticalSeam < this.verticalSeams.length; verticalSeam++)
         {
             verticalSeams[verticalSeam] = calculateVerticalSeam(newPixels);
-            newPixels = removeVerticalSeam(verticalSeams[verticalSeam], newPixels);
+            Seam seam = verticalSeams[verticalSeam];
+            newPixels = recalculate(removeVerticalSeam(seam, newPixels));
         }
         for (int horizontalSeam = 0; horizontalSeam < this.horizontalSeams.length; horizontalSeam++)
         {
             horizontalSeams[horizontalSeam] = calculateHorizontalSeam(newPixels);
```

Comment:
Because you remove vertical seams, that will change the horizontal energy. So the first time through this loop you aren't working with the correct energies.

Reply:
I'm not sure I understand what I did wrong with the for loop here

Reply:
I'm wrong here. I didn't realize that recalculate will do both horizontal and vertical energies.